updateStableIds/src/main/java/org/reactome/release/update_stable_ids/UpdateStableIds.java

```diff
@@ -0,0 +1,170 @@
+package org.reactome.release.update_stable_ids;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collector;
+import java.util.stream.Collectors;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.gk.model.GKInstance;
+import org.gk.model.ReactomeJavaConstants;
+
+import static org.gk.model.ReactomeJavaConstants.*;
+import org.gk.persistence.MySQLAdaptor;
+import org.gk.persistence.TransactionsNotSupportedException;
+import org.gk.schema.GKSchemaClass;
+import org.gk.schema.InvalidAttributeException;
+import org.reactome.release.common.database.InstanceEditUtils;
+
+public class UpdateStableIds {
+
+	private static final Logger logger = LogManager.getLogger();
+	
+	public static void stableIdUpdater(MySQLAdaptor dbaSlice, MySQLAdaptor dbaPrevSlice, MySQLAdaptor dbaGkCentral, Long personId) throws Exception {
+		
+		logger.info("Generating InstanceEdits for " + dbaSlice.getDBName() + " and " + dbaGkCentral.getDBName());
+		// Instance Edits for test_slice and gk_central
+		String creatorName = "org.reactome.release.update_stable_ids";
+		GKInstance sliceIE = InstanceEditUtils.createInstanceEdit(dbaSlice, personId, creatorName);
+		GKInstance gkCentralIE = InstanceEditUtils.createInstanceEdit(dbaGkCentral, personId, creatorName);
+
+		// At time of writing (December 2018), test_slice is a non-transactional database. This check has been put in place as a safety net in case that changes.
+		if (dbaSlice.supportsTransactions()) {
+			dbaSlice.startTransaction();
+		}
+		dbaGkCentral.startTransaction();
+		
+		//TODO:Should legacy regulation instances be updated?
+		//TODO:Ensure that re-run handling is implemented (backing up DB)
```

Comment:
I think this comment needs more explaining...?

Reply:
Actually it creates a 'test_slice_XX_snapshot', and it gets used only in the following release. Basically it will look at all instances in the current test_slice, and compare the number of 'Modified' attributes it has with the instance from the snapshot